mantidproject/mantid
# Add collimation length to to TOFSANSResolutionByPixel

Fixes #13213 
Fixes #11617

**Context**

Several changes to the existing QResolutionByPixel algorithm were performed:
- Addition of custom collimation length
- Refactoring of QResolutionCalculation in order to unit test it
- Fixes to the algorithm
- Updates to the documentation (rst file)
### For Testing

Pleas find the relevant test files here: \olympic\Babylon5\Scratch\Anton\Testing\13213_Add_Collimation_Length

There is a python script provided which will create the relevant test workspaces for you. You need to set the PATH_TO_DATA to the location of your data. The spcript produces:
- **wavSample**: A workspace which was saved out from a SANS2D reduction process just before passing it into Q1D. This will be the sample workspace definint the size
- **wavMod**: A moderator worksapce for TS2 instruments. 

**General**
-  Make sure that the unit tests pass
-  Please code review

**Test functionality with and without gravity**
1. Run the script to produce the two workspaces
2. Open the TOFSANSResolutionByPixel algorithm
3. Set R1 = 1 R2 = 1, deltaR = 0.1, CollimationLength = 4, SimgaModeator =modWav, CollimationLength = 5
4. Set the InputWorkspace to wavSample
5. Set the Moderator Workspace to the wavMod
6. Execute the algorithm
   - Confirm that there are no errors
7. Check the gravity option and set the extra length to 4
8. Execute the algorithm
   - Confirm that there are no errors

**Test wrong input**
1. Run the script to produce the two workspaces
2. Open the TOFSANSResolutionByPixel algorithm
3. Set R1 = 1 R2 = 1, deltaR = 0.1, CollimationLength = 4, CollimationLength = 5
4. Set the InputWorkspace to **wavMod**
5. Set the Moderator Workspace to the wavMod
6. Execute the algorithm
   - Confirm that an error is raised since wavMod does not contain an instrument
